test/test_replace.py

```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+
+import pytest
+
+from simple_parsing import replace, subgroups
+
+from .test_utils import TestSetup
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class A(TestSetup):
+    a: float = 0.0
+
+
+@dataclass
+class B(TestSetup):
+    b: str = "bar"
+    b_post_init: str = field(init=False)
+
+    def __post_init__(self):
+        self.b_post_init = self.b + "_post"
+
+
+@dataclass
+class AB(TestSetup):
+    integer_only_by_post_init: int = field(init=False)
+    integer_in_string: str = "1"
+    a_or_b: A | B = subgroups({"a": A, "b": B}, default="a")
+
+    def __post_init__(self):
+        self.integer_only_by_post_init = int(self.integer_in_string)
+
+
+@dataclass
+class C:
+    c: bool = False
+
+
+@dataclass
+class D:
+    d: int = 0
+
+
+@dataclass(frozen=True)
+class CD(TestSetup):
+    c_or_d: C | D = subgroups({"c": C, "d": D}, default="c")
+
+    other_arg: str = "bob"
+
+
+@dataclass
+class NestedDataclasses:
+    nested_arg: str = "nested_arg"
+    arg1: int = 0
+    arg2: float = 0.0
+
+
+@dataclass
+class OuterDataclass(TestSetup):
+    some_arg: str = "some_arg"
+    nested: NestedDataclasses = field(default_factory=NestedDataclasses)
+
+
+@pytest.mark.parametrize(
+    ("config", "changes"),
+    [
+        (A(a=2.0), {"a": 2.0}),
+        pytest.param(A(), {"b": 1.0}, marks=pytest.mark.xfail(reason="Unrecognized arguments")),
+        (B(b="test"), {"b": "test"}),
+        (AB(a_or_b=A()), {"a_or_b": "a"}),
+        (AB(a_or_b=B(b="foo")), {"__subgroups__@a_or_b": "b", "a_or_b": {"b": "foo"}}),
```

Comment:
I don't understand this here: Why would we allow these weird keys to be passed as an argument to `replace`?
They aren't fields of the dataclass, so in my opinion they should raise an error.

Feel free to tell me what you think, but I'm thinking that this doesn't look that natural to use. It also mixes up the expected behaviour of `dataclasses.replace` with some command-line-parsing-type logic. I don't think that's something we want `replace` to do.

Reply:
I mentioned this below.